source/non_matching/quadrature_generator.cc

```diff
@@ -1505,8 +1505,6 @@ namespace NonMatching
         const auto dof_handler_cell =
           cell->as_dof_handler_iterator(*dof_handler);
 
-        const FE_Q<dim> fe_q1(1);
```

Comment:
We should not create a `FE_Q` object in every call to `set_active_cell()`.

Reply:
Nice observation! Are you likely going to fall into the `if` branch where you now create the object more than once? If so, the object could also be made a member variable of the class so that it is really only created once.

Reply:
Only if one uses `hp` for the level set. With a member, we would be on the safe side.

Reply:
I have an even better solution. Make this a member variable of type `Lazy<FE_Q<dim>>` and then in the place where you're accessing it, use `fe.get(...)`. This way you create the object either never (if the FE_Q is not used at all) or on first use.

Reply:
Done. There is no doxygen docu showing up for `Lazy`, is this intentional?

Reply:
Look's like `Lazy` doesn't work because `FE_Q` is not moveable. Any suggestions? Locally this works fine on my machine (gcc 9.4.0).

Reply:
What does it take to make `FE_Q` movable? Just removing `const` from the polynomial degree in the base class?